src/carplay.cs

```diff
@@ -0,0 +1,719 @@
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPListSection : NSSecureCoding {
+
+		[Export ("initWithItems:header:sectionIndexTitle:")]
+		IntPtr Constructor (CPListItem [] items, [NullAllowed] string header, [NullAllowed] string sectionIndexTitle);
+
+		[Export ("initWithItems:")]
+		IntPtr Constructor (CPListItem [] items);
+
+		[NullAllowed, Export ("header")]
+		string Header { get; }
+
+		[NullAllowed, Export ("sectionIndexTitle")]
+		string SectionIndexTitle { get; }
+
+		[Export ("items", ArgumentSemantic.Copy)]
+		CPListItem [] Items { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (CPTemplate))]
+	[DisableDefaultCtor]
+	interface CPListTemplate : CPBarButtonProviding {
+
+		[Export ("initWithSections:")]
+		IntPtr Constructor (CPListSection [] sections);
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		ICPListTemplateDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; set; }
+
+		[Export ("sections", ArgumentSemantic.Copy)]
+		CPListSection [] Sections { get; }
+
+		[NullAllowed, Export ("title")]
+		string Title { get; set; }
+
+		[Export ("updateSections:")]
+		void UpdateSections (CPListSection [] sections);
+	}
+
+	interface ICPListTemplateDelegate { }
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject))]
+	interface CPListTemplateDelegate {
+
+		[Abstract]
+		[Export ("listTemplate:didSelectListItem:completionHandler:")]
+		void DidSelectListItem (CPListTemplate listTemplate, CPListItem item, Action completionHandler);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	interface CPManeuver : NSCopying, NSSecureCoding {
+
+		[NullAllowed, Export ("symbol", ArgumentSemantic.Strong)]
+		UIImage Symbol { get; set; }
+
+		[Export ("instructionVariants", ArgumentSemantic.Copy)]
+		string [] InstructionVariants { get; set; }
+
+		[NullAllowed, Export ("distanceFromPreviousManeuver", ArgumentSemantic.Copy)]
+		NSMeasurement<NSUnitLength> DistanceFromPreviousManeuver { get; set; }
+
+		[NullAllowed, Export ("userInfo", ArgumentSemantic.Strong)]
+		NSObject UserInfo { get; set; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPMapButton : NSSecureCoding {
+
+		[Export ("initWithHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([NullAllowed] Action<CPMapButton> handler);
+
+		[Export ("enabled")]
+		bool Enabled { [Bind ("isEnabled")] get; set; }
+
+		[Export ("hidden")]
+		bool Hidden { [Bind ("isHidden")] get; set; }
+
+		[NullAllowed, Export ("image", ArgumentSemantic.Strong)]
+		UIImage Image { get; set; }
+
+		[NullAllowed, Export ("focusedImage", ArgumentSemantic.Strong)]
+		UIImage FocusedImage { get; set; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (CPTemplate))]
+	interface CPMapTemplate : CPBarButtonProviding {
```

Comment:
should not the base .ctor be copied here ?
along with the disable default ctor attribute ?

Reply:
@spouliot mmm what base ctor?

```
[Abstract]
[NoWatch, NoTV, NoMac, iOS (12,0)]
[BaseType (typeof (NSObject))]
interface CPTemplate : NSSecureCoding {
}
```

Reply:
I misread (see later comment) but the default .ctor might not be something to expose in every (or all) subclasses